src/steps/format-json.js

```diff
@@ -0,0 +1,53 @@
+import dasherize from "dasherize";
+import camelize from "../util/camelize";
+import Resource from "../types/Resource";
+import Collection from "../types/Collection";
+
+export function dasherizeResource(resource, registry) {
+  let exceptions = registry.behaviors(resource.type).dasherizeOutput.exceptions;
+  return transformResourceKeys(resource, dasherize, exceptions);
+}
+
+export function dasherizeResourceOrCollection(toDasherize, registry) {
+  return transformResourceOrCollection(toDasherize, dasherizeResource, registry);
+}
+
+export function camelizeResource(resource, registry) {
+  let exceptions = registry.behaviors(resource.type).dasherizeOutput._inverseExceptions;
+  return transformResourceKeys(resource, camelize, exceptions);
+}
+
+export function camelizeResourceOrCollection(toCamelize, registry) {
+  return transformResourceOrCollection(toCamelize, camelizeResource, registry);
+}
+
+function transformResourceKeys(resource, transformFn, exceptions) {
+  for (let key in resource.attrs) {
+    let transformedKey = exceptions.hasOwnProperty(key) ? exceptions[key] : transformFn(key);
+    if (transformedKey !== key) {
+      resource.attrs[transformedKey] = resource.attrs[key];
+      delete resource.attrs[key];
+    }
+  }
+
+  return resource;
+}
+
+function transformResourceOrCollection(toTransform, transformFn, registry) {
+  if (toTransform instanceof Collection) {
```

Comment:
One of my least favorite things about the current codebase is how many `instanceof` checks like this it has and needs. They all result, I think, from the fact that I don't have a powerful enough abstraction that can handle this type juggling in one place. When I do a big refactor in August, I plan to find a better abstraction—probably some monadic type that stores a collection and an `isSingleResource` flag. For now, though, my one attempt at extracting out these checks is the `mapResources` function in the `type-handling` utils. You should be using that here, as it'll take care of this check for you.